core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -946,15 +955,25 @@ private Builder(TableMetadata base) {
       this.previousFileLocation = base.metadataFileLocation;
       this.previousFiles = base.previousFiles;
       this.refs = Maps.newHashMap(base.refs);
-      this.statisticsFiles =
```

Comment:
Currently one stats file per snapshot. But I think in future it may track multiple files too. 

I followed the same pattern as existing puffin files. I want to keep the interfaces consistent.